src/Compilers/CSharp/Portable/BoundTree/BoundCollectionExpression.cs

```diff
@@ -3,19 +3,45 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class BoundCollectionExpression
     {
         private partial void Validate()
         {
+#if DEBUG
+            var collectionCreation = this.CollectionCreation;
+            while (collectionCreation is BoundConversion conversion)
+                collectionCreation = conversion.Operand;
+
+            Debug.Assert(collectionCreation
+                            is null
+                            or BoundObjectCreationExpression
+                            or BoundCall
+                            or BoundNewT
```

Comment:
>BoundNewT [](http://example.com/codeflow?start=31&length=9)

Based on behavior of ```getConstructArguments```, we do not expect collectionCreation to be set for this case. Let's assert that